### PR TITLE
Fix adduct missing charge

### PR DESCRIPTION
**Pull request by niekdejonge**
Still many adducts are not recognized since they do not have a charge e.g. [M+H+Na] would not be used, since the charge is unknown. Before some of these were repaired, because we stored common adduct types. In addition now the charge stored in spectrum.get("charge") is used to repair an adduct that misses the overall charge.

### CODE REVIEW

File: matchms/filtering/metadata_processing/derive_ionmode.py [hunk 1]
```diff
@@ -29,20 +29,20 @@ def derive_ionmode(spectrum_in: SpectrumType) -> SpectrumType:
...
-
-    adduct = spectrum.get("adduct", None)
-    # Harmonize adduct string
-    if adduct:
-        adduct = _clean_adduct(adduct)
-
     ionmode = spectrum.get("ionmode")
     if ionmode:
         assert ionmode == ionmode.lower(), ("Ionmode field not harmonized.",
                                             "Apply 'make_ionmode_lowercase' filter first.")
```
Comment (florian-huber) [inline matchms/filtering/metadata_processing/derive_ionmode.py]:
```suggestion
        assert ionmode == ionmode.lower(), "Ionmode field not harmonized."
```

Comment (florian-huber) [reply matchms/filtering/metadata_processing/derive_ionmode.py]: `make_ionmode_lowercase` no longer exists since this is now part of the `Metadata` class.

Comment (niekdejonge) [reply matchms/filtering/metadata_processing/derive_ionmode.py]: I guess we can remove this assert statement altogether in this case.

File: matchms/filtering/metadata_processing/derive_ionmode.py [hunk 2]
```diff
@@ -30,8 +30,6 @@ def derive_ionmode(spectrum_in: SpectrumType) -> SpectrumType:
     spectrum = spectrum_in.clone()
 
     ionmode = spectrum.get("ionmode")
-    if ionmode:
-        assert ionmode == ionmode.lower(), "Ionmode field not harmonized."
```
Comment (florian-huber) [inline matchms/filtering/metadata_processing/derive_ionmode.py]: You are right, can be removed entirely.

File: matchms/filtering/metadata_processing/derive_ionmode.py [hunk 3]
```diff
@@ -51,5 +48,4 @@ def derive_ionmode(spectrum_in: SpectrumType) -> SpectrumType:
```
Comment (florian-huber) [inline matchms/filtering/metadata_processing/derive_ionmode.py:48]: I think ionmode should be set to `None` instead of `"n/a"`.

Comment (florian-huber) [reply matchms/filtering/metadata_processing/derive_ionmode.py:48]: (we haven't touched this filter in a while it seems...)

**Review by florian-huber — APPROVED** [review]
Looks good. `derive_ionmode` has long not been touched it seems. I would say we should remove missing entries, while this currently still sets ionmode to "n/a".

But OK, data without known ionmode is still not a problem that we should spend too much on in `matchms` 😄

**Pull request merged**